Source/Private/ASDisplayNodeInternal.h

```diff
@@ -55,6 +55,18 @@ typedef NS_OPTIONS(NSUInteger, ASDisplayNodeMethodOverrides)
   ASDisplayNodeMethodOverrideClearFetchedData   = 1 << 6
 };
 
+typedef NS_OPTIONS(uint_least32_t, ASDisplayNodeAtomicFlags)
+{
+  ASDNFlagSynchronous = 1 << 0,
+  ASDNFlagLayerBacked = 1 << 1,
+};
+
+#define checkFlag(flags, flag) ((flags.load() & flag) == flag)
+#define setFlag(flags, flag, x) (x ? flags.fetch_or(flag) : flags.fetch_and(~flag))
+
+#define atomic_isSynchronous checkFlag(_atomicFlags, ASDNFlagSynchronous)
+#define atomic_setIsSynchronous(x) setFlag(_atomicFlags, ASDNFlagSynchronous, x)
```

Comment:
Let's eliminate the `flags` argument from `checkFlag` and `setFlag`, and then remove the `atomic_*` macros. 
Let's also have `setFlag` return the old value of the flag, so it would shake out like this:

```objc
#define checkFlag(flag) ((_atomicFlags.load() & flag) == flag)
// Returns the old value of the flag as a BOOL.
#define setFlag(flag, x) (((x ? _atomicFlags.fetch_or(flag) : _atomicFlags.fetch_and(~flag)) & flag) != 0)
 
// In code:
BOOL sync = checkFlag(ASDNFlagSynchronous);
setFlag(ASDNFlagSynchronous, YES);
```